eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml

```diff
@@ -3,7 +3,15 @@ steps:
     - pwsh: |
         $filesInCommonDir = git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only -- 'eng/common/*'
 
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and ($LASTEXITCODE -eq 0) -and ($filesInCommonDir.Count -gt 0))
+        # Find the default branch of the repo
+        $defaultBranchLine = git remote show origin | Where-Object {$_ -Like "*HEAD branch:*"}
```

Comment:
```
$defaultBranch = (git symbolic-ref --short refs/remotes/origin/HEAD) -replace "origin/"
```

Reply:
I tried that earlier, but unfortunately it does not work on the pipeline. Probably because it always checks-out to a detached head.
https://dev.azure.com/azure-sdk/playground/_build/results?buildId=738474&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=24311d03-63ec-5098-c8e5-7df09da2b6d4

Reply:
Ah yeah that is definitely an interesting case. Let me see if I can figure out another way because I'd like to avoid parsing the git output.

Reply:
Perhaps?

```
((Get-Content ./.git/refs/remotes/origin/HEAD) -split ' ')[1]
```

Reply:
Unfortunately I as not able to find another solution that works consistency so I do think we will have to parse the output of `git remote show`. However I do think we can do this parsing in a little easier way without weird splits and magic indexing.

```
$defaultBranch = (git remote show origin | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
```
- Out-string makes all the lines go into one multiline string
- [`(?ms)`](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options) means match across multiple lines and match newlines as any char.
- `$1` replace the entire string with the match between `(\w+)`

Reply:
FYI @mitchdenny @sima-zhu after spending some time investigating I believe that PS snippet is the easiest way that I could find to get the default branch name in cases where we need the branch but don't want to hardcode it.